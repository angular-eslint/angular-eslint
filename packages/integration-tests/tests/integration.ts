import * as execa from 'execa';
import path from 'path';

const FIXTURES_DIR = path.join(__dirname, '../fixtures/');

function normalizeOutput(value: string): string {
  return value.replace(new RegExp(FIXTURES_DIR, 'g'), '__ROOT__/');
}

function runLint(directory: string): unknown {
  try {
    const cwd = path.join(FIXTURES_DIR, directory);
    process.chdir(cwd);

    execa.sync(path.join(cwd, '../../install-fixture-deps.sh'), {
      cwd,
    });

    const { stdout: lintOutput } = execa.sync('npx', ['ng', 'lint'], {
      cwd,
    });

    return normalizeOutput(lintOutput);
  } catch (error) {
    return normalizeOutput(error.stdout || error);
  }
}

const integrationTests: [string][] = [['angular-cli-workspace']];

describe.each(integrationTests)('%s', (directory) => {
  it('it should produce the expected lint output', () => {
    const lintOutput = runLint(directory);
    expect(lintOutput).toMatchSnapshot();
  });
});
