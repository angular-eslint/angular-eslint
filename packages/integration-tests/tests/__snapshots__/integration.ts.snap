// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`angular-cli-workspace it should produce the expected lint output 1`] = `
"
Linting \\"angular-cli-workspace\\"...

__ROOT__/angular-cli-workspace/src/app/app.component.html
  1:31  error  Invalid binding syntax. Use [(expr)] instead  @angular-eslint/template/banana-in-a-box

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the \`--fix\` option.


__ROOT__/angular-cli-workspace/src/app/app.component.ts
   2:1  error    'rxjs/Rx' import is restricted from being used. Please import directly from 'rxjs' instead                             no-restricted-imports
   5:1  error    Unexpected property on console object was called                                                                       no-restricted-syntax
  15:3  warning  Lifecycle interface 'OnInit' should be implemented for method 'ngOnInit'. (https://angular.io/styleguide#style-09-01)  @angular-eslint/use-lifecycle-interface

✖ 3 problems (2 errors, 1 warning)


__ROOT__/angular-cli-workspace/src/app/example.pipe.ts
  6:8  error  Classes decorated with @Pipe decorator should implement PipeTransform interface  @angular-eslint/use-pipe-transform-interface

✖ 1 problem (1 error, 0 warnings)


__ROOT__/angular-cli-workspace/src/app/example.component.ts
   4:13  error  Strings must use singlequote                                                                         quotes
  12:3   error  Use @Input rather than the \`inputs\` metadata property (https://angular.io/styleguide#style-05-12)    @angular-eslint/no-inputs-metadata-property
  13:3   error  Use @Output rather than the \`outputs\` metadata property (https://angular.io/styleguide#style-05-12)  @angular-eslint/no-outputs-metadata-property
  14:3   error  Use @Output rather than the \`host\` metadata property (https://angular.io/styleguide#style-05-12)     @angular-eslint/no-host-metadata-property
  18:3   error  In the class \\"ExampleComponent\\", the output property \\"onFoo\\" should not be prefixed with on          @angular-eslint/no-output-on-prefix
   6:35  error  Invalid binding syntax. Use [(expr)] instead                                                         @angular-eslint/template/banana-in-a-box
   8:15  error  Invalid binding syntax. Use [(expr)] instead                                                         @angular-eslint/template/banana-in-a-box
   8:29  error  Invalid binding syntax. Use [(expr)] instead                                                         @angular-eslint/template/banana-in-a-box
   9:48  error  Invalid binding syntax. Use [(expr)] instead                                                         @angular-eslint/template/banana-in-a-box

✖ 9 problems (9 errors, 0 warnings)
  5 errors and 0 warnings potentially fixable with the \`--fix\` option.


__ROOT__/angular-cli-workspace/src/app/app.component.spec.ts
  19:6  error  Strings must use singlequote  quotes

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the \`--fix\` option.
"
`;
